src/nsls2api/middleware.py

```diff
@@ -0,0 +1,27 @@
+from starlette.types import ASGIApp, Message, Receive, Scope, Send
+from starlette.datastructures import MutableHeaders
+import time
+
+
+class ProcessTimeMiddleware:
+    app: ASGIApp
+
+    def __init__(self, app: ASGIApp) -> None:
+        self.app = app
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if scope["type"] != "http":
+            await self.app(scope, receive, send)
+            return
+        start_time = time.time()
+
+        async def send_wrapper(message: Message) -> None:
+            if message["type"] == "http.response.start":
+                headers = MutableHeaders(scope=message)
+                end_time = time.time()
```

Comment:
```suggestion
                end_time = time.monotonic()
```

Reply:
As this is for benchmarking, I've gone with `time.perf_counter()` to get the best accuracy.